test/e2e/declarative_test_fwk/helper.py

```diff
@@ -176,23 +195,85 @@ def _assert_conditions(self, expectations: dict, reference: k8s.CustomResourceRe
 
             else:
                 raise Exception(f"Condition {condition_name} is provided with invalid value: {expected_value} ")
 
-    def assert_items(self, expectations: dict, state: dict) -> None:
-        """Asserts state against supplied expectations
-        Override it as needed for custom verifications
+    def assert_items_aws(self, expectations: dict, k8s_resource: dict) -> None:
+        """Asserts boto3 response against supplied expectations
 
         Args:
-            expectations: dictionary with items (expected) to assert in state
-            state: dictionary with items (actual)
+            expectations: dictionary with items (expected) to assert in k8s resource
+            k8s_resource: the current status/spec of the k8s resource
+
+        Returns:
+            None
+        """
+        if not expectations:
+            # nothing to assert as there are no expectations
+            return
+        if not k8s_resource:
+            # there are expectations but no given k8s resource state to validate
+            # following assert will fail and assert introspection will provide useful information for debugging
+            assert expectations == k8s_resource
+
+        resource_name = ""
+        for (key, value) in k8s_resource.items():
+            if key == "name":
+                resource_name = value
+        # get aws resource
+        aws_resource = self.get_aws_resource(resource_name)
+        latestTags = self.mdb.list_tags(ResourceArn=aws_resource.get("ARN"))['TagList']
+
+        for (key, value) in expectations.items():
+            if key == "Tags":
+                self.assert_tags(value, latestTags)
+                continue
+            # validation for specific fields
+            if self.assert_extra_items_aws(aws_resource, key, value):
+                continue
+            assert value == aws_resource.get(key)
+
+    def get_aws_resource(self, resource_name):
+        """get aws resource from boto3
+        Override it for each resource type
+
+        Args:
+            k8s_resource: the current status/spec of the k8s resource
+
+        Returns:
+            aws resource
+        """
+        return
+
+    def assert_extra_items_aws(self, aws_resource, expected_name, expected_value):
+        """Asserts specific field boto3 response against supplied expectation
+        Override it as needed for validations of specific field
+
+        Args:
+            mdb: memorydb client from boto3
+            aws_resource: specific resource from aws memorydb
+            k8s_resource: the current status/spec of the k8s resource
+            expected_name: expected name of a specific field to assert in aws resource
+            expected_value: expected value of a specific field to assert in aws resource
+
+        Returns:
+            True/False
```

Comment:
nit: Explain why you would return true. eg. `Return true if the expected_name was asserted successfully, false if it was not asserted`

Reply:
Added explanation.

Reply:
I don't see them?

Reply:
I added above in the comment part not in returns part.
```
If resource has expected_name as a field, execute custom logic to validate that specific field and return true.
Otherwise, if resource doesn't have expected_name as a field, return false
```

Reply:
def assert_extra_items_aws(self, expected_name, expected_value, aws_resource) -> bool:
       """Asserts specific fields that boto3 response against supplied expectation
       Override it as needed for validations of specific field
       If resource has expected_name as a field, execute custom logic to validate that specific field and return true
       Otherwise, if resource doesn't have expected_name as a field, return false
       Args:
           expected_name: expected name of a specific field to assert in aws resource
           expected_value: expected value of a specific field to assert in aws resource
           aws_resource: specific resource from aws memorydb
       Returns:
           True/False
       """
       return False

Reply:
Oh ok. Usually we include the comments about the return value on the same line as where we define its type. This is the "Google docstring format" which it seems like the original author of this framework was following. Here are the docs for examples on using it - https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
Modified the words base on @jljaco comment https://github.com/aws-controllers-k8s/memorydb-controller/pull/65#discussion_r1122265736

Reply:
Modified the format!